### PR TITLE
feat: add header for skipping the blockcache

### PR DESCRIPTION
**Pull request by aschmahmann**
Adds a header (protected by the Authorization header) for skipping the blockcache to better aid debugging.

Based off of #143

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ The following emojis are used to highlight certain changes:
 * 🛠 - BREAKING CHANGE.  Action is required if you use this functionality.
 * ✨ - Noteworthy change to be aware of.
 
+## Unreleased
+
+### Added
+
+- Added support for a debugging header `Rainbow-No-Blockcache` that is gated by the `Authorization` header and does not use the local block cache for the request
```
Comment (aschmahmann) [inline CHANGELOG.md:18]: Any thoughts on where the docs for this header should go?

File: handlers.go [hunk 2]
```diff
@@ -218,13 +221,33 @@ func setupGatewayHandler(cfg Config, nd *Node, tracingAuth string) (http.Handler
 			if request.Header.Get("Tracestate") != "" {
 				request.Header.Del("Tracestate")
 			}
+			if request.Header.Get(NoBlockcacheHeader) != "" {
+				request.Header.Del(NoBlockcacheHeader)
+			}
+		}
+
+		// Process cache skipping header
+		if noBlockCache := request.Header.Get(NoBlockcacheHeader); noBlockCache != "" {
```
Comment (aschmahmann) [inline handlers.go:230]: Thoughts on the text that should go here? Should it just be `true` or `false`?

File: handlers.go [hunk 3]
```diff
@@ -218,13 +221,33 @@ func setupGatewayHandler(cfg Config, nd *Node, tracingAuth string) (http.Handler
...
+			request = request.WithContext(newCtx)
 		}
+
 		prevHandler.ServeHTTP(writer, request)
 	})
 
 	return handler, nil
 }
 
+const NoBlockcacheHeader = "Rainbow-No-Blockcache"
```
Comment (aschmahmann) [inline handlers.go:247]: Up for better suggestions on the header name

File: handlers.go [hunk 4]
```diff
@@ -218,13 +221,33 @@ func setupGatewayHandler(cfg Config, nd *Node, tracingAuth string) (http.Handler
...
 				request.Header.Del("Tracestate")
 			}
+			if request.Header.Get(NoBlockcacheHeader) != "" {
+				request.Header.Del(NoBlockcacheHeader)
+			}
+		}
+
+		// Process cache skipping header
+		if noBlockCache := request.Header.Get(NoBlockcacheHeader); noBlockCache != "" {
+			ds, err := leveldb.NewDatastore("", nil)
```
Comment (aschmahmann) [inline handlers.go:231]: This bring in leveldb as a dependency, but IIRC the levelDB memory datastore behaves a bunch better under parallel block writing/fetching load compared to the mapdatastore + sync datastore wrapper. I don't expect this to be used a ton either way, so if people disagree I'm ok switching to the more basic datastore.

File: CHANGELOG.md (hunk 1 above)
Comment (lidel) [reply CHANGELOG.md:18]: Maybe create `docs/HEADERS.md` ?

File: handlers.go (hunk 2 above)
Comment (lidel) [reply handlers.go:230]: Perhaps enable on explicit `true`, disable on everything else?
```suggestion
		if noBlockCache := request.Header.Get(NoBlockcacheHeader); noBlockCache == "true" {
```

File: handlers.go (hunk 3 above)
Comment (lidel) [reply handlers.go:247]: Maybe a generic `Rainbow-Skip-Caches`? 
We may want to skip things like namesys cache as well at some point too.

**Comment by lidel** [comment]
(other things landed, I'm going to rebase and review)

File: handlers.go (hunk 3 above)
Comment (lidel) [reply handlers.go:247]: After sleeping on this, having an explicit flag for each system is more useful for debugging, let's keep as-is.

**Review by lidel — APPROVED** [review]
Lgtm, thank you for fixing the prometheus panic in tests.

Pushed small refactor (moved configuration to Config an related `with*` funcs), and also made sure default (when `RAINBOW_TRACING_AUTH` is unset) behavior has explicit test.

Merging, will release next.

**Pull request merged**